crates/oxc_linter/src/rules/unicorn/no_useless_length_check.rs

```diff
@@ -0,0 +1,303 @@
+    #[diagnostic(
+        severity(warning),
+        help("The empty check is useless as `Array#every()` returns `true` for an empty array.")
+    )]
+    Every(#[label] Span),
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessLengthCheck;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// It checks for an unnecessary array length check in a logical expression
+    /// The cases are:
+    ///  array.length === 0 || array.every(Boolean) (array.every returns true if array is has elements)
+    ///  array.length > 0 && array.some(Boolean) (array.some returns false if array is empty)
+    ///
+    /// ### Why is this bad?
+    /// An extra unnecessary length check is done
+    ///
+    /// ### Example
+    /// ```javascript
+    /// if(array.length === 0 || array.every(Boolean)){
+    ///    do something!
+    /// }
+    ///
+    /// ```
+    NoUselessLengthCheck,
+    correctness
+);
+
+struct ConditionDTO<T: ToString> {
+    property_name: T,
+    binary_operators: Vec<BinaryOperator>,
+}
+
+fn get_static_member_property_name<'a>(expr: Option<&'a MemberExpression<'a>>) -> Option<&'a str> {
+    match expr {
+        Some(MemberExpression::StaticMemberExpression(expr)) => Some(expr.property.name.as_str()),
+        _ => None,
+    }
+}
+
+fn is_useless_check<'a>(
+    left: &'a Expression<'a>,
+    right: &'a Expression<'a>,
+    operator: LogicalOperator,
+) -> Option<NoUselessLengthCheckDiagnostic> {
+    let every_condition = ConditionDTO {
+        property_name: "every",
+        binary_operators: vec![BinaryOperator::StrictEquality],
+    };
+    let some_condition = ConditionDTO {
+        property_name: "some",
+        binary_operators: vec![BinaryOperator::StrictInequality, BinaryOperator::GreaterThan],
+    };
+
+    let mut array_name: &str = "";
+    let active_condition = {
+        if operator == LogicalOperator::Or {
+            every_condition
+        } else {
+            some_condition
+        }
+    };
+    let mut binary_expression_span: Option<Span> = None;
+    let mut call_expression_span: Option<Span> = None;
+
+    let l = match left.without_parenthesized() {
+        Expression::BinaryExpression(expr) => {
+            let Expression::MemberExpression(left_expr) = expr.left.get_inner_expression() else {
+                return None;
+            };
+            array_name = left_expr.object().get_identifier_reference()?.name.as_str();
+            binary_expression_span = Some(expr.span);
+
+            active_condition.binary_operators.contains(&expr.operator)
+                && left_expr.is_specific_member_access(array_name, "length")
+                && expr.right.is_raw("0")
+        }
+        Expression::CallExpression(expr) => {
+            array_name =
+                expr.callee.get_member_expr()?.object().get_identifier_reference()?.name.as_str();
+            let property_name = get_static_member_property_name(expr.callee.get_member_expr())?;
```

Comment:
does `expr.callee.get_member_expr()?.static_property_name` work?

Reply:
Seems that it works fine, so I will use it